tests/scripts/local_monitoring_tests/rest_apis/test_local_monitoring_e2e_workflow.py

```diff
@@ -0,0 +1,211 @@
+"""
+Copyright (c) 2024, 2024 Red Hat, IBM Corporation and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import copy
```

Comment:
@shreyabiradar07 - Can you share the test results. Also update the test documentation.

Reply:
Test results: 
[local_monitoring_e2e_tests_openshift.zip](https://github.com/user-attachments/files/16705570/e2e_tests_openshift.zip)
Jekins local monitoring functional tests Minikube: [104](https://ci.app-svc-perf.corp.redhat.com/job/ExternalTeams/job/Autotune/job/kruize_functional_tests/104/#showFailuresLink ) Openshift: [106](https://ci.app-svc-perf.corp.redhat.com/job/ExternalTeams/job/Autotune/job/kruize_functional_tests/106/)

Reply:
@shreyabiradar07 - Can you check why this test failed on openshift with the below error:

```
assert [<ValidationError: "'cpu' is a required property">, <ValidationError: "'cpu' is a required property">, <ValidationErro... property">, <ValidationError: "'cpu' is a required property">, <ValidationError: "'cpu' is a required property">, ...] == ''
```

Reply:
@chandrams as there are no requests and limits set for tfb workloads, current config is blank and the `[<ValidationError: "'cpu' is a required property">]`, error is due to `CPU Usage is less than a millicore, No CPU Recommendations can be generated` notification as cpu is specified as a required field in the `list_recomm_json_schema`

example for reference:

```
 "monitoring_start_time": "2024-08-21T13:04:06.000Z",
                      "recommendation_engines": {
                        "cost": {
                          "pods_count": 1,
                          "confidence_level": 0.0,
                          "config": {
                            "limits": {
                              "memory": {
                                "amount": 3.65719552E7,
                                "format": "bytes"
                              }
                            },
                            "requests": {
                              "memory": {
                                "amount": 3.65719552E7,
                                "format": "bytes"
                              }
                            }
                          },
                          "variation": {
                            "limits": {
                              "memory": {
                                "amount": 3.65719552E7,
                                "format": "bytes"
                              }
                            },
                            "requests": {
                              "memory": {
                                "amount": 3.65719552E7,
                                "format": "bytes"
                              }
                            }
                          },
                          "notifications": {
                            "323001": {
                              "type": "notice",
                              "message": "CPU Usage is less than a millicore, No CPU Recommendations can be generated",
                              "code": 323001
                            }
                          }
                        },
```